kratos/solving_strategies/strategies/residualbased_newton_raphson_strategy.h

```diff
@@ -879,6 +879,37 @@ class ResidualBasedNewtonRaphsonStrategy
         KRATOS_CATCH("");
     }
 
+    /**
+     * @brief Gets the current solution vector
+     * @param rDofSet The set of degrees of freedom (DOFs)
+     * @return A vector containing the current solution values for the given DOF set
+     * @details This method retrieves the current solution values for the provided DOF set. Each value is accessed using the equation ID associated with each DOF.
+     */
+    Vector GetCurrentSolution(DofsArrayType& rDofSet){
```

Comment:
I think we should pass a reference in here. Otherwise we are creating a copy at return time.

Reply:
Reference is now passed to avoid copies